src/main/java/org/apache/commons/collections4/MapBuilder.java

```diff
@@ -0,0 +1,115 @@
+package org.apache.commons.collections4;
+
+import org.apache.commons.collections4.map.HashedMap;
+
+import java.util.*;
+
+/**
+ * Defines an Helper Builder that generates a {@code Map}
+ * A Builder class to help decide which type of map to use based on simple requirements.
+ * Currently It takes care of only basic types of Maps and can be extended to different types of Maps available in the ecosystem.
+ *
+ * <pre>{@code
+ * Map builderMap = new MapBuilder().setIterationOrder(MapBuilder.KeyOrder.INSERTION_ORDER).build();
+ * builderMap.put("A", 1);
+ * builderMap.put("X", 24);
+ * builderMap.put("B", 2);
+ * builderMap.put("Y", 26);
+ * }</pre>
+ *
+ * @author Amita Pradhan
+ */
+public class MapBuilder {
+
+    private Comparator comparator;
+    private KeyOrder iterationOrder;
+    private boolean synchronizedMap;
+    private boolean immutable;
+    private Map data;
+
+    public MapBuilder() {
```

Comment:
I only see this useful if you cover all Map implementations in the JDK. It's really the only way IMO to see if this builder is handy or not but I am somewhat doubtful in general. Let's see where this goes from here... For, me it's simpler to use the right class in the right place. For example, how do I build a `SortedMap<Foo, Bar>` and is returned to the call site as a `SortedMap<Foo, Bar>`?

Reply:
Will try to add as many as possible :-). Wanted to have something like this when I started using Maps and tried to search generateMap in MapUtils for help.

Reply:
Hi @Amita-Pradhan ,

This file is missing the Apache license header. Note that the GitHub build fails (see Actions tab).

Reply:
Done. Fixed the header check and checkstyle errors.